vulnerabilities data

lielbin1 opened the pull request:
Download the vulnerabilities for a specific year to create functions for the vulnerabilities data

Shayaomer committed:
Merge branch 'master' into 1-find_list_of_installed_programs

Shayaomer committed:
Added conditional create of nvd folder

If nvd doesnt exist, create the folder and download the json files, if it exists do nothing

lielbin1 committed:
update sevetrity for cve

lielbin1 committed:
severity of cve

lielbin1 committed:
Merge remote-tracking branch 'origin/master'

amitrezilion commented on main.py:
```diff
@@ -1,5 +1,6 @@
 import utils
 from cvesCollection import CVEsCollection
+from cve import Cve
 
```
utils: this file isn't really utils. create a class that downloads the db and allows retrieving the downloaded db (per year, like the util function)

danaSror replied:
done

amitrezilion commented on cve.py:
```diff
@@ -5,16 +5,23 @@ def __init__(self, dict : {}):
+
 
     def getCveId(self) -> int:
         """
```
What's the purpose of the comments in this file?

danaSror replied:
remove this comments

amitrezilion commented on cve.py:
```diff
@@ -5,16 +5,23 @@ def __init__(self, dict : {}):
     def getCveId(self) -> int:
         """
 
         :return:
```
Read about dataclasses and use it, instead of this class

amitrezilion commented on cve.py:
```diff
@@ -5,16 +5,23 @@ def __init__(self, dict : {}):
+        self.identifier = dict['cve']['CVE_data_meta']['ID']
         self.ASSIGNER = dict['cve']['CVE_data_meta']['ASSIGNER']
         self.description = dict['cve']['problemtype']['problemtype_data'][0]['description'][0]['value']
+        self.severity = dict['impact']
```
This logic shouldn't be in a simple data structure class. should get the severity without knowing about v2 or v3

amitrezilion commented on main.py:
```diff
@@ -8,12 +9,13 @@ def main():
-    cve_year_collection.getCveByIndex(0)
+    cve = cve_year_collection.getCveByIndex(51)
     print("Hello")
 
```
CVEsCollection: Instead of printing the json, should create a function that returns the string, and then the caller will print it. (what if instead of printing it, we'll want to write to a file)

amitrezilion commented on main.py:
```diff
@@ -8,12 +9,13 @@ def main():
-    cve_dict = cve_year_collection.getCVE_Items()
-    cve_year_collection.getCveByIndex(0)
+    cve = cve_year_collection.getCveByIndex(51)
     print("Hello")
```
CVEsCollection: why do u save the number of cves (instead of just return len(cves)) or something?

amitrezilion commented on main.py:
```diff
@@ -8,12 +9,13 @@ def main():
     print("Hello")
 
 
 
```
CVEsCollection: You didn't change the file according to the talk we had last sunday. let's talk about it again :)

amitrezilion commented on cve.py:
```diff
@@ -30,3 +37,14 @@ def getCveDescription(self) -> str:
+        return self.severity
+
+    def printCve(self):
+        print("CVE identifier: " + self.identifier)
```
Should not have printing in a logical class. if you want to print data, the caller should print it.

Shayaomer committed:
some resolves

shimonShouei committed:
logging

Shayaomer committed:
CR resolves

Made a new class of 'installedSoftware' and
connected the new class to the logger.

shimonShouei committed:
Merge branch 'master' into 1-find_list_of_installed_programs

shimonShouei committed:
Merge pull request #2 from shimonShouei/1-find_list_of_installed_programs

Merge branch 'master' into 1-find_list_of_installed_programs
